packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -90,13 +98,69 @@ export enum SCOPE_TYPES {
   GLOBAL = 'global'
 }
 
-export class ScopeContainer {
+export abstract class VariableContainer {
+  public variables: Map<String, VariableContainer>;
+
+  public constructor(
+    variables: Map<String, VariableContainer> = new Map<
+      String,
+      VariableContainer
+    >()
+  ) {
+    this.variables = variables;
+  }
+
+  public getAllVariables(): ApexVariable[] {
+    const result: ApexVariable[] = [];
+    this.variables.forEach(container => {
+      const avc = container as ApexVariableContainer;
+      result.push(
+        new ApexVariable(avc.name, avc.value, avc.type, avc.variablesRef)
+      );
+    });
+    return result;
```

Comment:
This is used in scopesRequest and variablesRequest to get the relevant variables. This will go through every variable in the container and create a VSCode Debugger Variable object